### PR TITLE
Update posm with tests

### PR DESCRIPTION
**Pull request by dianakocsis**

### CODE REVIEW

File: contracts/base/Dispatcher.sol [hunk 1]
```diff
@@ -291,13 +291,13 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, V4SwapRout
 
     /// @notice Function to be used instead of msg.sender, as the contract performs self-reentrancy and at
     /// times msg.sender == address(this). Instead _msgSender() returns the initiator of the lock
-    /// @dev overrides BaseActionsRouter._msgSender in V4Router
-    function _msgSender() internal view override returns (address) {
+    function _msgSender() internal view returns (address) {
```
Comment (hensha256) [inline contracts/base/Dispatcher.sol:294]: i think we should remove this now and just use `msgSender` although gas will go up a little

Comment (dianakocsis) [reply contracts/base/Dispatcher.sol:294]: kk

Comment (dianakocsis) [reply contracts/base/Dispatcher.sol]: didn't make gas go up at all - i think u had this in a separate pr where we were discussing the same thing

File: contracts/base/Dispatcher.sol [hunk 2]
```diff
@@ -281,23 +281,18 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, V4SwapRout
     /// @return output The resultant recipient for the command
     function map(address recipient) internal view returns (address) {
         if (recipient == Constants.MSG_SENDER) {
```
Comment (hensha256) [inline contracts/base/Dispatcher.sol:283]: Once we have `main` in can you delete `MSG_SENDER` `ADDRESS_THIS` and `CONTRACT_BALANCE` from `Constants.sol` and just use the ones that are in `ActionsConstants.sol` in v4?

File: test/integration-tests/V3ToV4Migration.test.ts [hunk 3]
```diff
@@ -1032,8 +1032,8 @@ describe('V3 to V4 Migration Tests:', () => {
         '0x',
       ])
 
-      v4Planner.addAction(Actions.SETTLE_WITH_BALANCE, [USDC.address])
-      v4Planner.addAction(Actions.SETTLE_WITH_BALANCE, [WETH.address])
+      v4Planner.addAction(Actions.SETTLE, [USDC.address, 1, false])
+      v4Planner.addAction(Actions.SETTLE, [WETH.address, 1, false])
```
Comment (hensha256) [inline test/integration-tests/V3ToV4Migration.test.ts:1036]: Then these tests can have these constants that are already in constants.ts
1 -> `CONTRACT_BALANCE`
false -> `SOURCE_ROUTER`

Comment (dianakocsis) [reply test/integration-tests/V3ToV4Migration.test.ts]: since CONTRACT_BALANCE and OEPN_DELTA now switched, i used OPEN_DELTA instead

Comment (hensha256) [reply test/integration-tests/V3ToV4Migration.test.ts]: wait what

Comment (hensha256) [reply test/integration-tests/V3ToV4Migration.test.ts]: OH yeah youre right

**Review by hensha256 — APPROVED** [review]

**Pull request merged**